src/policy/policy.cpp

```diff
@@ -141,9 +141,12 @@ bool IsStandardTx(const CTransaction& tx, const std::optional<unsigned>& max_dat
         else if ((whichType == TxoutType::MULTISIG) && (!permit_bare_multisig)) {
             reason = "bare-multisig";
             return false;
-        } else if (IsDust(txout, dust_relay_fee)) {
+        } else if (whichType != TxoutType::ANCHOR && IsDust(txout, dust_relay_fee)) {
             reason = "dust";
             return false;
+        } else if (whichType == TxoutType::ANCHOR && !permit_ephemeral_anchor) {
+            reason = "ephemeral-anchor"; return false;
+            return false;
```

Comment:
there's no actual use case for 2+, so this would neatly apply onto my branch against master which *does* restrict it via policy. I could port more logic over to make these checks, of course.

Reply:
I guess I was thinking that `CountEphmeralOutputs` should just be `bool HasEphemeralOutput` and the `too-many-ephmeral-anchors` check should be here, same as `multi-op-return`.

Reply:
done. also added 0-fee check in `PreChecks`